### PR TITLE
feat(content-releases): remove cron when scheduled release is deleted

### PR DESCRIPTION
**Pull request by Feranchz**
### What does it do?

Remove the cron job when a scheduled release is delete. 

### Enabling the future flags

The scheduling functionality is implemented through future flags. To begin, ensure the future flag is set to true. Navigate to `config/features.js` and include `contentReleasesScheduling` as an option within your future object. Create the future object if it does not exist.

Example:
```js
// config/features.js

module.exports = ({ env }) => ({
  future: {
    contentReleasesScheduling: true,
  },
});

```

### How to test it

Note: You need to get all the scheduledJobs from scheduling service (There is no a way in UI to see this, so you would need to go to the console)

1. Create a release with a scheduled date.
2. scheduledJobs should have the job for publish the just created release.
3. Delete the release
4. scheduledJobs should not have the job for publish the deleted release

### CODE REVIEW

File: packages/core/content-releases/server/src/services/__tests__/release.test.ts [hunk 1]
```diff
@@ -430,6 +436,46 @@ describe('release service', () => {
...
+
+      // @ts-expect-error Ignore missing properties
+      const releaseService = createReleaseService({ strapi: strapiMock });
+
+      await releaseService.delete(1);
+
+      expect(mockSchedulingCancel).toHaveBeenCalledWith(1);
+    });
+
+    it('does not remove the scheduling if the release is not scheduled', async () => {
```
Comment (simotae14) [inline packages/core/content-releases/server/src/services/__tests__/release.test.ts:460]: same here
Do you mean 
```
it('does not remove the scheduling if the release is not deleted', async () => {
```
?

Comment (Feranchz) [reply packages/core/content-releases/server/src/services/__tests__/release.test.ts:460]: No, in both cases the release is deleted. But a deleted release could be scheduled or not.

File: packages/core/content-releases/server/src/services/__tests__/release.test.ts [hunk 2]
```diff
@@ -430,6 +436,46 @@ describe('release service', () => {
 
       expect(() => releaseService.delete(1)).rejects.toThrow('Release already published');
     });
+
+    it('remove the scheduling if the release is scheduled', async () => {
```
Comment (remidej) [inline packages/core/content-releases/server/src/services/__tests__/release.test.ts:440]:
```suggestion
    it('removes the scheduling if the release is scheduled', async () => {
```

**Review by remidej — COMMENTED** [review]
I haven't QAed but the code looks good to me 👍

**Review by simotae14 — APPROVED** [review]
I need to QA it but it looks ok to me

**Pull request merged**